prov/rxm/src/rxm_cq.c

```diff
@@ -383,11 +414,11 @@ ssize_t rxm_handle_recv_comp(struct rxm_rx_buf *rx_buf)
 		       "queue\n");
 		rx_buf->unexp_msg.addr = match_attr.addr;
 		rx_buf->unexp_msg.tag = match_attr.tag;
-		dlist_insert_tail(&rx_buf->unexp_msg.entry, &recv_queue->unexp_msg_list);
-		fastlock_release(&recv_queue->lock);
+		dlist_insert_tail(&rx_buf->unexp_msg.entry, &rx_buf->recv_queue->unexp_msg_list);
+		fastlock_release(&rx_buf->recv_queue->lock);
 		return 0;
 	}
-	fastlock_release(&recv_queue->lock);
+	fastlock_release(&rx_buf->recv_queue->lock);
```

Comment:
Please move the changes in this file to a separate patch.

Reply:
Done. the changes have been moved to the 3rd commit

Reply:
I'm fine with the changes to this commit, but I wonder if it actually helps.  Even if the compiler keeps the stack variable in the release build (it may not), the trade-off here is that we need another layer of indirection to access recv_queue.

Reply:
I've analyzed produced code by compilers.
This helps to produce a slightly optimized code for GNU compiler 7.x. But for Intel compiler 15.0 there are no any effects.